### PR TITLE
feat(jsonnet): add JSONNET_PATH environement variable

### PR DESCRIPTION
**Pull request by chaudum**
`<CWD>/lib` and `<CWD>/vendor` are common jsonnet library paths.
The jsonnet language server can use the `JSONNET_PATH` env variable to
resolve referenced files within these folders.

Signed-off-by: Christian Haudum <christian.haudum@gmail.com>

### CODE REVIEW

File: lua/lspconfig/jsonnet_ls.lua [hunk 1]
```diff
@@ -1,9 +1,19 @@
 local configs = require 'lspconfig/configs'
 local util = require 'lspconfig/util'
 
+-- common jsonnet library paths
+local function jsonnet_path()
+  local paths = {
+    vim.fn.getcwd() .. '/lib',
+    vim.fn.getcwd() .. '/vendor',
```
Comment (lithammer) [inline lua/lspconfig/jsonnet_ls.lua:8]: Shouldn't this be relative to the resolved `root_dir` rather than `cwd`?

Comment (mjlbach) [reply lua/lspconfig/jsonnet_ls.lua:8]: Also use path.join instead of string concatenation.

File: lua/lspconfig/jsonnet_ls.lua [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
+
 configs.jsonnet_ls = {
   default_config = {
     cmd = { 'jsonnet-language-server' },
+    cmd_env = { JSONNET_PATH = vim.fn.getcwd() }, -- will be overwritten by `on_new_config`
     filetypes = { 'jsonnet', 'libsonnet' },
-    root_dir = function(fname)
-      return util.root_pattern 'jsonnetfile.json'(fname) or util.find_git_ancestor(fname)
-    end,
+    root_dir = vim.fn.getcwd(), -- will be overwritten by `on_new_config`
```
Comment (mjlbach) [inline lua/lspconfig/jsonnet_ls.lua:18]: Keep the original root dir

**Review by mjlbach — APPROVED** [review]

File: lua/lspconfig/jsonnet_ls.lua [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
+    on_new_config = function(new_config, file_dir)
+      local root_dir = util.root_pattern('jsonnetfile.json')(file_dir) or util.find_git_ancestor(file_dir)
+      new_config.cmd_env = {
+        JSONNET_PATH = jsonnet_path(root_dir)
+      }
+      new_config.root_dir = root_dir
```
Comment (mjlbach) [inline lua/lspconfig/jsonnet_ls.lua:24]: Why do you have to replace root dir in on_new_config?

**Review by mjlbach — CHANGES_REQUESTED** [review]

File: lua/lspconfig/jsonnet_ls.lua [hunk 4]
```diff
@@ -10,18 +10,20 @@ local function jsonnet_path(root_dir)
...
     cmd = { 'jsonnet-language-server' },
     cmd_env = { JSONNET_PATH = vim.fn.getcwd() }, -- will be overwritten by `on_new_config`
     filetypes = { 'jsonnet', 'libsonnet' },
-    root_dir = vim.fn.getcwd(), -- will be overwritten by `on_new_config`
+    root_dir = get_root_dir,
     on_new_config = function(new_config, file_dir)
-      local root_dir = util.root_pattern('jsonnetfile.json')(file_dir) or util.find_git_ancestor(file_dir)
       new_config.cmd_env = {
-        JSONNET_PATH = jsonnet_path(root_dir)
+        JSONNET_PATH = jsonnet_path(get_root_dir(file_dir))
```
Comment (mjlbach) [inline lua/lspconfig/jsonnet_ls.lua:25]: Why do you have to do this again? file_dir is root_dir (I would rename this)

Comment (chaudum) [reply lua/lspconfig/jsonnet_ls.lua:25]: What I get as root_dir is the directory of the opened file :shrug:

Comment (mjlbach) [reply lua/lspconfig/jsonnet_ls.lua:25]: then the root pattern is failing to find a root and you should get the same result again

Comment (mjlbach) [reply lua/lspconfig/jsonnet_ls.lua:25]: Or there is a bug, this is where it's supposed to be passed 

https://github.com/mjlbach/nvim-lspconfig/blob/c1af9bd8296b1627ed0517b1915c76545ea7cee9/lua/lspconfig/configs.lua#L208-L209

Comment (chaudum) [reply lua/lspconfig/jsonnet_ls.lua]: You're right. I had an erroneous override of the default config in the setup function :man_facepalming:

**Comment by chaudum** [comment]
@mjlbach Should I rebase and squash commits?

File: lua/lspconfig/jsonnet_ls.lua [hunk 5]
```diff
@@ -1,13 +1,28 @@
...
+    util.path.join(root_dir, 'lib'),
+    util.path.join(root_dir, 'vendor'),
+  }
+  return table.concat(paths, ':')
+end
+
 configs.jsonnet_ls = {
   default_config = {
     cmd = { 'jsonnet-language-server' },
+    cmd_env = { JSONNET_PATH = vim.fn.getcwd() }, -- will be overwritten by `on_new_config`
```
Comment (mjlbach) [inline lua/lspconfig/jsonnet_ls.lua:16]: Why does this need to be a placeholder?

**Comment by mjlbach** [comment]
Need to run stylua in the root of lspconfig

**Pull request merged**